perfkitbenchmarker/linux_benchmarks/dpb_distcp_benchmark.py

```diff
@@ -0,0 +1,159 @@
+                  'File System to use as destination of the distcp operation')
+
+flags.DEFINE_integer('file_size_mbs', 10,
+                     'File size to use for the distcp source files')
+
+flags.DEFINE_integer('num_files', 10, 'Number of distcp source files')
+
+FLAGS = flags.FLAGS
+
+SUPPORTED_DPB_BACKENDS = [dpb_service.DATAPROC, dpb_service.EMR]
+
+
+def GetConfig(user_config):
+  return configs.LoadConfig(BENCHMARK_CONFIG, user_config, BENCHMARK_NAME)
+
+
+def CheckPrerequisites(benchmark_config):
+  """Verifies that the required resources are present.
+
+  Raises:
+    perfkitbenchmarker.errors.Config.InvalidValue: On encountering invalid
+    configuration.
+  """
+  dpb_service_type = benchmark_config.dpb_service.service_type
+  if dpb_service_type not in SUPPORTED_DPB_BACKENDS:
+    raise errors.Config.InvalidValue('Invalid backend for distcp. Not in:{}'.
+                                     format(str(SUPPORTED_DPB_BACKENDS)))
+
+
+def Prepare(benchmark_spec):
+  del benchmark_spec  # Unused.
+  pass
+
+
+def Run(benchmark_spec):
+  """Runs distributed_copy benchmark and reports the results.
+
+  Args:
+    benchmark_spec: Spec needed to run the distributed synth benchmark
+
+
+  Returns:
+    A list of samples
+  """
+  run_uri = benchmark_spec.uuid.split('-')[0]
+  source = '/{}'.format(run_uri)
+  update_source_default_fs = False
+
+  if FLAGS.distcp_source_fs != BaseDpbService.HDFS_FS:
+    source = '{}:/{}'.format(FLAGS.distcp_source_fs, source)
+    benchmark_spec.dpb_service.CreateBucket(source)
+    update_source_default_fs = True
+
+  source_dir = '{}{}'.format(source, '/dfsio')
+  source_data_dir = '{}{}'.format(source_dir, '/io_data')
+
+  # TODO(saksena): Respond to data generation failure
+  benchmark_spec.dpb_service.generate_data(source_dir, update_source_default_fs,
+                                           FLAGS.num_files,
+                                           FLAGS.file_size_mbs)
+
+  destination_dir = '/{}{}'.format(run_uri, '/dfsio_destination')
+  if FLAGS.distcp_dest_fs != BaseDpbService.HDFS_FS:
+    destination_dir = '{}:/{}'.format(FLAGS.distcp_source_fs, destination_dir)
+
+  start = datetime.datetime.now()
+  benchmark_spec.dpb_service.distributed_copy(source_data_dir, destination_dir)
+  end_time = datetime.datetime.now()
+  run_time = (end_time - start).total_seconds()
+
+  results = []
+  metadata = copy.copy(benchmark_spec.dpb_service.GetMetadata())
+  metadata.update({'source_fs': FLAGS.distcp_source_fs})
+  metadata.update({'destination_fs': FLAGS.distcp_dest_fs})
+  metadata.update({'num_files': FLAGS.num_files})
+  metadata.update({'file_size_mbs': FLAGS.file_size_mbs})
+  if FLAGS.zones:
+    zone = FLAGS.zones[0]
+    region = zone.rsplit('-', 1)[0]
+    metadata.update({'regional': True})
+    metadata.update({'region': region})
+  elif FLAGS.cloud == 'AWS':
+    metadata.update({'regional': True})
+    metadata.update({'region': 'aws_default'})
+
```

Comment:
Generally, regional: False should never happen, since it does not offer a fair comparison against AWS which is always regional, which is why I did not include it